### DISCUSSION TITLE
How to add sourcemaps for dev?

### DISCUSSION THREAD

**Discussion by theogravity** [alex8088/electron-vite#133] [Q&A]
If an error gets thrown, the stack trace references the `out/main/index.js` file which is the bundle, vs the original typescript file.

**Comment by alex8088** [comment]
set `build.sourcemap` option to `true` or `inline`

**Reply by theogravity** [reply]
That does generate the sourcemap, but the stack trace doesn't use it. I've read that you can start node with [--enable-source-maps](https://nodejs.org/dist/latest/docs/api/cli.html#cli_enable_source_maps), but not sure how to do that with `electron-vite`.

**Reply by Armaldio** [reply]
In addition to the `build.sourcemap` option, like @theogravity said, you can turn on the flag using the environment variable:

`NODE_OPTIONS=--enable-source-maps electron-vite dev -w`

I'm using it like this inside my package.json script:
```
...
"dev": "NODE_OPTIONS=--enable-source-maps electron-vite dev -w",
...
```

**Reply by alex8088** [reply]
https://electron-vite.org/guide/cli.html#options

 `electron-vite dev --sourcemap`

**Reply by Armaldio** [reply]
When I tried, this option had no effect

**Reply by alex8088** [reply]
soucemap file in output dir
